server/src/main/java/org/opensearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -395,19 +399,21 @@ public Query termsQuery(List<?> values, QueryShardContext context) {
                 if (!context.keywordFieldIndexOrDocValuesEnabled()) {
                     return super.termsQuery(values, context);
                 }
-                BytesRef[] bytesRefs = new BytesRef[values.size()];
-                for (int i = 0; i < bytesRefs.length; i++) {
-                    bytesRefs[i] = indexedValueForSearch(values.get(i));
+                BytesRef[] iBytesRefs = new BytesRef[values.size()];
```

Comment:
This change does not look right: we only need to the rewrite for doc values when doc values are enabled ( hasDocValues()).

Reply:
I don't quite understand what you mean, could you please describe more detail?

Reply:
Yeah, non-issue here :) sorry about the confusion